pkg/server/node_auth.go

```diff
@@ -0,0 +1,111 @@
+package server
+
+import (
+	"fmt"
+
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/kube/kclient"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/security"
+	"istio.io/istio/pkg/spiffe"
+	"istio.io/istio/pkg/util/sets"
+	"istio.io/istio/security/pkg/server/ca"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ClusterNodeAuthorizer is a component that implements a subset of Kubernetes Node Authorization
```

Comment:
@paulwilljones does Istio have a similar ClusterNodeAuthorizer in their own implementation?

Reply:
Yes, [`ClusterNodeAuthorizer`](https://github.com/istio/istio/blob/1.22.1/security/pkg/server/ca/node_auth.go#L74) is the same, however, I've reimplemented it here to make sure the pod informer is synced before the index is created.

Reply:
Could you add a link to that implementation in this file?